lib/timers.js

```diff
@@ -169,7 +169,11 @@ function listOnTimeout() {
     // Check if this loop iteration is too early for the next timer.
     // This happens if there are more timers scheduled for later in the list.
     if (diff < msecs) {
-      this.start(msecs - diff, 0);
+      var timeRemaining = msecs - (TimerWrap.now() - timer._idleStart);
```

Comment:
I've thought about this a little more and now that the current PR code just makes timers work properly without any explicit handling for specific bugs, it would be hard to throw in extra comments without it becoming confusing _(in my opinion)_.  I defer to your judgement but that's my opinion.